pkg/etcd/etcd_test.go

```diff
@@ -0,0 +1,291 @@
+package etcd
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/rancher/k3s/pkg/clientaccess"
+	"github.com/rancher/k3s/pkg/daemons/config"
+	"github.com/rancher/k3s/pkg/util/tests"
+	"github.com/robfig/cron/v3"
+	etcd "go.etcd.io/etcd/clientv3"
+)
+
+func generateTestConfig() *config.Control {
+	return &config.Control{
+		HTTPSPort:             6443,
+		SupervisorPort:        6443,
+		ClusterDomain:         "cluster.local",
+		DataDir:               "/tmp/k3s/", // Different than the default value
+		FlannelBackend:        "vxlan",
+		EtcdSnapshotName:      "etcd-snapshot",
+		EtcdSnapshotCron:      "0 */12 * * *",
+		EtcdSnapshotRetention: 5,
+		EtcdS3Endpoint:        "s3.amazonaws.com",
+		EtcdS3Region:          "us-east-1",
+		SANs:                  []string{"127.0.0.1"},
+	}
+}
+
+func generateTestHandler() http.Handler {
```

Comment:
Would this also be a good candidate to be put in the test package for use elsewhere?

Reply:
I did not feel like there was enough other functions in the pkg that required a handler as an argument. Thus I did not move it into runtime.go to ensure that the tests pkg was a slim as possible.

Reply:
We can leave here for now until we see how if it'd be useful elsewhere and in the RKE2 codebase.